File: app/src/main/java/es/uplace/uplace/domain/enumeration/ApartmentType.java
package es.uplace.uplace.domain.enumeration;

public enum ApartmentType {
    // HOUSES, RURALS, FLATS, APARTMENTS, TOWERS, LOFTS
    // MAPPING (Not necessary or may be yes...?)
    HOUSES("houses"), RURALS("rurals"), FLATS("flats"), APARTMENTS("apartments"), TOWERS("towers"), LOFTS("lofts");

    private String apartmentType;

    ApartmentType(String apartmentType) {
        this.apartmentType = apartmentType;
    }

    public String getTypeName() {
        return apartmentType;
    }

    public static ApartmentType fromTypeName(String name) {
        switch (name) {
            case "houses":
                return ApartmentType.HOUSES;
            case "rurals":
                return ApartmentType.RURALS;
            case "flats":
                return ApartmentType.FLATS;
            case "apartments":
                return ApartmentType.APARTMENTS;
            case "towers":
                return ApartmentType.TOWERS;
            case "lofts":
                return ApartmentType.LOFTS;
            default:
                throw new IllegalArgumentException("ApartmentType [" + name
                    + "] not supported.");
        }
    }

}

File: app/src/main/java/es/uplace/uplace/domain/enumeration/ApartmentType.kt
package es.uplace.uplace.domain.enumeration

enum class ApartmentType(val typeName: String) {
    // HOUSES, RURALS, FLATS, APARTMENTS, TOWERS, LOFTS
    // MAPPING (Not necessary or may be yes...?)
    HOUSES("houses"),
    RURALS("rurals"),
    FLATS("flats"),
    APARTMENTS("apartments"),
    TOWERS("towers"),
    LOFTS("lofts");


    companion object {
        fun fromTypeName(name: String): ApartmentType {
            when (name) {
                "houses" -> return ApartmentType.HOUSES
                "rurals" -> return ApartmentType.RURALS
                "flats" -> return ApartmentType.FLATS
                "apartments" -> return ApartmentType.APARTMENTS
                "towers" -> return ApartmentType.TOWERS
                "lofts" -> return ApartmentType.LOFTS
                else -> throw IllegalArgumentException("ApartmentType [" + name + "] not supported.")
            }
        }
    }

}
